k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapResponseParser.java

```diff
@@ -412,6 +411,22 @@ private String parseQuoted() throws IOException {
         throw new IOException("parseQuoted(): end of stream reached");
     }
 
+    private String parseQuotedOrNil() throws IOException {
```

Comment:
I didn't give it much thought. The other one is potentially public API, i.e. if we ever decide to release an IMAP library. It's also an exceptional case for these response data holder classes.